actionpack/lib/action_controller/metal/rack_delegation.rb

```diff
@@ -8,9 +8,15 @@ module RackDelegation
     delegate :headers, :status=, :location=, :content_type=,
              :status, :location, :content_type, :response_code, :to => "@_response"
 
-    def dispatch(action, request)
+    module ClassMethods
```

Comment:
Yeah, it's a method to create a new controller instance in a given rack environment.
Since the environment is a rack territory,  this is the best file I could think of to put this method in.

I'll change name/file if you wish.